### PR TITLE
samples: matter: Fixed BLE advertising when not in pairing mode.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Currently for the Device Firmware Upgrade purposes nrfconnect
devices may advertise over BLE even with Thread enabled and not
in the pairing mode. However they should not advertise CHIPoBLE
service, but the one used for DFU (SMP).

* Added dfu_over_smp module handling performing DFU over BLE using
SMP protocol.
* Changed light_bulb config to start pairing mode autostart on boot,
as it is no longer forbidden by the spec.

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: samples/matter/light_bulb/README.rst [hunk 1]
```diff
@@ -189,9 +189,8 @@ Commissioning the device
     :start-after: matter_door_lock_sample_commissioning_start
     :end-before: matter_door_lock_sample_commissioning_end
 
-Before starting the commissioning procedure, you must make the device discoverable over Bluetooth LE.
-To enable Bluetooth LE for a predefined period of time (15 minutes by default), press **Button 4**.
-If the Bluetooth LE advertising times out, you can re-enable using the same button.
+Before starting the commissioning procedure, the device must be discoverable over Bluetooth LE that starts automatically upon the device startup, but only for a predefined period of time (15 minutes by default).
```
Comment (greg-fer) [inline samples/matter/light_bulb/README.rst:192]:
```suggestion
Before starting the commissioning procedure, the device must be made discoverable over Bluetooth LE, which starts automatically upon the device startup, but only for a predefined period of time (15 minutes by default).
```

File: samples/matter/light_bulb/README.rst [hunk 2]
```diff
@@ -189,9 +189,8 @@ Commissioning the device
...
+If the Bluetooth LE advertising times out, you can re-enable pressing the **Button 4**.
```
Comment (greg-fer) [inline samples/matter/light_bulb/README.rst:193]:
```suggestion
If the Bluetooth LE advertising times out, you can re-enable it by pressing **Button 4**.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]
Two minor comments.

**Review by greg-fer — APPROVED** [review]

File: doc/nrf/releases/release-notes-latest.rst [hunk 3]
```diff
@@ -145,6 +145,8 @@ The following list summarizes the most important changes inherited from the upst
 
   * Support for Certificate-Authenticated Session Establishment (CASE) for communication among operational Matter nodes.
   * Support for OpenThread's DNS Client to enable Matter node discovery on Thread devices.
+  * Fixed the known issue KRKNWK-10387 where Matter service was needlessly advertised over Bluetooth LE during DFU.
+    Now if Matter pairing mode is not opened and the Bluetooth LE advertising is needed due to DFU requirements, the SMP service is advertised.
```
Comment (Damian-Nordic) [inline doc/nrf/releases/release-notes-latest.rst:149]:
```suggestion
    Now if Matter pairing mode is not opened and the Bluetooth LE advertising is needed due to DFU requirements, only the SMP service is advertised.
```

File: samples/matter/weather_station/src/app_task.cpp [hunk 4]
```diff
@@ -204,13 +176,15 @@ void AppTask::PostEvent(AppEvent::Type type, AppEvent::Handler handler)
...
+void AppTask::RequestSMPAdvertisingStart(void)
 {
-	/* For now just print update progress and confirm data chunk without any additional checks. */
-	LOG_INF("Software update progress %d B / %d B", offset, size);
-
-	return 0;
+	AppEvent event;
+	event.mType = AppEvent::Type::kStartSMPAdvertising;
+	event.mHandler = [](AppEvent *) { GetDFUOverSMP().StartBLEAdvertising(); };
+	sAppTask.PostEvent(&event);
```
Comment (Damian-Nordic) [inline samples/matter/weather_station/src/app_task.cpp:185]: We have a helper for that:

```suggestion
	sAppTask.PostEvent(AppEvent::Type::kStartSMPAdvertising, [](AppEvent *) { GetDFUOverSMP().StartBLEAdvertising(); });
```

**Review by Damian-Nordic — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Pull request merged**